### ISSUE TITLE
No normalization layer?

### ISSUE THREAD

**Issue by GH920** [swz30/MPRNet#37]
Great work! I'm trying to rebuild the network with tensorflow framework. However, all blocks in MRPNet have no normalization layer, neither BatchNormalization nor GroupNormalization. As the network architecture is pretty huge, I just wonder whether the network exists the gradient vanishing/exploding issue in training. Especially for the ORS-Net, it contains 8 CAB blocks.
I have roughly built the MPRNet, but the training is not stable. The loss is easily becoming inf or nan.

**Comment by swz30** [comment]
Hi @GH920 

It is common in Image Restoration to not use any normalization as it hurts the performance. We did not experience stability issues with our experimental settings.

**Comment by swz30** [comment]
Hi @GH920

It is common in Image Restoration to not use any normalization as it hurts the performance. We did not experience stability issues with our experimental settings. I would recommend using a low initial learning rate. Please check [this](https://github.com/swz30/MPRNet/issues/26).

**Comment by GH920** [comment]
Thanks for your explanation. I'll try with your recommendation.

**Comment by swz30** [comment]
Hi @GH920, 

In case you are still having unstable training issues, try adding gradient clipping: https://github.com/swz30/MPRNet/issues/51#issuecomment-843802843
